com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -67,11 +63,40 @@ public StackingSensor(ISensor wrapped, int numStackedObservations)
             }
 
             // TODO support arbitrary stacking dimension
-            m_Shape[0] *= numStackedObservations;
-            m_StackedObservations = new float[numStackedObservations][];
-            for (var i = 0; i < numStackedObservations; i++)
+            m_Shape[m_Shape.Length - 1] *= numStackedObservations;
+
+            if (m_WrappedSensor.GetCompressionType() == SensorCompressionType.None)
             {
-                m_StackedObservations[i] = new float[m_UnstackedObservationSize];
+                m_StackedObservations = new float[numStackedObservations][];
+                for (var i = 0; i < numStackedObservations; i++)
+                {
+                    m_StackedObservations[i] = new float[m_UnstackedObservationSize];
+                }
+            }
+            else
+            {
+                m_StackedCompressedObservations = new byte[numStackedObservations][];
+                m_EmptyCompressedObservation = CreateEmptyPNG();
+                for (var i = 0; i < numStackedObservations; i++)
+                {
+                    m_StackedCompressedObservations[i] = m_EmptyCompressedObservation;
+                }
+
+                // Construct compression mapping
+                var compressibleSensor = m_WrappedSensor as ICompressibleSensor;
```

Comment:
You need to handle the case where the wrapped sensor isn't actually an ICompressibleSensor, just an ISensor. I think in that case, you should either leave the mapping empty, or the "identity" mapping (`mapping[i] == i`)

Reply:
This is inside the else statement of checking `m_WrappedSensor.GetCompressionType()` which should have a compression type other than None. My intention is to make all sensors that can have compressed observation being ICompressibleSensor, so it should always be ICompressibleSensor here.
Or should I check and raise an exception here if the sensor has `m_WrappedSensor.GetCompressionType() !=SensorCompressionType.None` but its' not an ICompressibleSensor?

Reply:
You shouldn't assume in general that `m_WrappedSensor.GetCompressionType() !=SensorCompressionType.None` means that it's an `ICompressibleSensor`. I think maybe a better name would change how we think about this - how about something like `ISparseChannelSensor`?

> My intention is to make all sensors that can have compressed observation being ICompressibleSensor, so it should always be ICompressibleSensor here.

I actually think this might make things harder for backwards compatibility reasons. I'll start another thread on that.

Reply:
This still needs to be addressed - if wrapping a non-ICompressibleSensor, assume an identity mapping.